src/gui/addnewtorrentdialog.cpp

```diff
@@ -530,9 +530,19 @@ void AddNewTorrentDialog::updateDiskSpaceLabel()
         }
     }
 
+    const Path path = m_ui->savePath->selectedPath();
+
+    qint64 freeSpace = Utils::Fs::freeDiskSpaceOnPath(path);
+    if (freeSpace < 0)
+    {
+        // for non-existent directories (which will be created on demand) `Utils::Fs::freeDiskSpaceOnPath`
+        // will return invalid value so instead query its parent path
+        freeSpace = Utils::Fs::freeDiskSpaceOnPath(path.parentPath());
```

Comment:
What if parent path doesn't exist too but only more distant ancestor?

Reply:
Then I suppose I should use a loop to query the ancestors. Will update.

Reply:
Done.